python/ray/actor.py

```diff
@@ -169,6 +169,25 @@ def invocation(args, kwargs):
         return invocation(args, kwargs)
 
 
+class ActorMetaClass(type):
+    """Metaclass used to prevent users from inheriting from an ActorClass."""
+
+    def __new__(cls, name, bases, attr):
+        for base in bases:
+            if type(base) is ActorMetaClass:
+                raise Exception("Attempted to define subclass '{}' of Ray "
```

Comment:
A `TypeError` seems more in line with python semantics here, compare the error you get when you do
```
class Class(Exception, dict): 
    pass
```

Reply:
Sure, I used `Exception` because that's what we were using below when users try to directly construct an actor object. I'll update both.

Reply:
On second thought, I'll leave the other as it's much less standard